src/main/java/com/example/nlern/controllers/subloadController.java

```diff
@@ -1,23 +1,143 @@
+        //List<User> users= UsRepo.findTop5ByOrderByIdDesc();
+        return  TestRepo.findTop5ByAuthorIdOrderByIdDesc(teacherId);
+    }
+
+
+
+    @PostMapping("tests")
+    public ResponseEntity createClient( @RequestBody FullTest fullTest) {//throws URISyntaxException {
+        //Client savedClient = clientRepository.save(client);
+        //return ResponseEntity.created(new URI("/clients/" + savedClient.getId())).body(savedClient);
+
+        int teacherId = 3; //------------- temp imitation until auth not added
+
+        Test test=  (fullTest.getTest()).get();
+        List<Question> questions=  (fullTest.getQuestions());
+
+        test.setAuthor(UsRepo.findById(teacherId).get());
+        test = TestRepo.save(test);
+
+
+        for (Question q: questions) {
+            q.setTest(test);
+            QuestRepo.save(q);
+
+        }
+
+
+        return ResponseEntity.ok(test);
+    }
+
+    @PutMapping("tests/{id}")
+    public ResponseEntity updateTest(@PathVariable Integer id, @RequestBody FullTest fullTest) {
+
+        Test test = TestRepo.findById(id).orElseThrow(RuntimeException::new);
+        Test receivedTest=  (fullTest.getTest()).get();
+        List<Question> receivedQuestions=  (fullTest.getQuestions());
+
+        List<Integer> createdQuestions =fullTest.getAdded();
+        List<Integer> deletedQuestions =fullTest.getDeleted();
+
+
+
+        test.setName(receivedTest.getName());
+        test.setDuration(receivedTest.getDuration());
+        test = TestRepo.save(test);
+
+
+
+        for (int i: deletedQuestions) {
+            // need add if(exist)
+            QuestRepo.deleteById(i);
+            continue;
+        }
+
+
+        for (Question q: receivedQuestions) {
+
+            if(  createdQuestions.contains(q.getId()) ){
+                q.setTest(test);
+                QuestRepo.save(q);
+                continue;
+            }
+
+
+            Question oldQuestion= QuestRepo.findById(q.getId()).get();
+            oldQuestion.setAnswers(q.getAnswers());
+            //oldQuestion.setAnswers(q.getAnswers());
+            QuestRepo.save(q);
+        }
+
+
+        return ResponseEntity.ok(test);
+    }
+
+
+    @GetMapping("tests/{id}")
+    public FullTest getTest(Model model, @PathVariable Integer id){
+        int teacherId = 3; //------------- temp imitation until auth not added
+        //List<User> users= UsRepo.findTop5ByOrderByIdDesc();
+        List<Question> questions=QuestRepo.findByTestIdOrderById(id);
+        Optional<Test> test =TestRepo.findById(id);
+
+        FullTest fullTest = new FullTest();
+        fullTest.setTest(test);
+        fullTest.setQuestions(questions);
+        //Map<String, Object> map = new HashMap<>();
+        //map.put("test", test);
+        //map.put("questions", questions);
+
+        return  fullTest;
+    }
+
+
+    @GetMapping("/t")
+    public List<Group> homePage(){
+        return (List<Group>) GroupRepo.findAll();
```

Comment:
create abstract method findAll() in repository that return List instead of Iterable